src/events/broker.py

```diff
@@ -196,7 +196,7 @@ def _on_start(self, event: StartEvent | PebbleReadyEvent) -> None:
         self.charm.on.update_status.emit()
 
         # only log once on successful 'on-start' run
-        if isinstance(self.charm.unit.status, ActiveStatus):
+        if self.healthy:
             logger.info(f'Broker {self.charm.unit.name.split("/")[1]} connected')
```

Comment:
**todo:** I think this is a downgrade from the original behavior found in the `update-status` handler.
Before, we have `update-status` do:

- ready-to-start
- healthy
- machine-configured
- broker-active

All of those would (incorrectly, admittedly) set status, which would be collected here. High time-cost calls like `broker-active` would only be executed after all the other conditions were evaluated.

Can we here check `self.unit_statuses`, maybe for whether it's not just a single-item set of `ACTIVE` or something? As I believe during that execution, the `emit()`'d update-status would have modified that charm class attr.
What do you think?

Reply:
I agree with you, this is not good as is. Maybe we can get away with looking at the last status in `self.unit_status`, and explicitly append `status.ACTIVE` (cf. previous thread)

Reply:
Agreed, happy to go for that :)

Reply:
Alrighty, fixed in b58a177